lib/puppet/file_system.rb

```diff
@@ -348,6 +349,21 @@ def self.path_string(path)
     @impl.path_string(path)
   end
 
+  # Produces a string representation of the opaque path handle, with expansions
+  # performed on ~.  For Windows, this means that C:\Users\Admini~1\AppData will
+  # be expanded to C:\Users\Administrator\AppData.  On POSIX filesystems, the
+  # value ~ will be expanded to something like /Users/Foo
+  #
+  # This method exists primarlily to resolve a Ruby deficiency where
+  # File.expand_path doesn't handle ~ in each segment on a Windows path
+  #
+  # @param path [Object] a path handle produced by {#pathname}
+  # @return [String] a string representation of the path
+  #
+  def self.expanded_path_string(path)
```

Comment:
Also on n*x we don't want to expand ~ either. Thus for n*x there is nothing to do but to return the original. (Hence: this is not the same as expand_path)